curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/use-recursion-to-create-a-range-of-numbers.english.md

```diff
@@ -0,0 +1,97 @@
+---
+id: 5cc0bd7a49b71cb96132e54c
+title: Use Recursion to Create a Range of Numbers
+challengeType: 1
+---
+
+## Description
+<section id='description'>
+
+In a [previous challenge](/learn/javascript-algorithms-and-data-structures/basic-javascript/replace-loops-using-recursion) you learned how to use recursion to replace a for loop. Now, let's look a more complex function that returns an array of consecutive integers starting with <code>1</code> through the number passed to the function.
+
+As mentioned in the previous challenge, there will be a <dfn>base case</dfn>.  The base case tells the recursive function when it no longer needs to call itself.  It is a simple case where the return value is already known. There will also be a <dfn>recursive call</dfn> which executes the original function with different arguments. If the function is written correctly, eventually the base case will be reached.
+
+For example, say you want to write a recursive function that returns an array containing the numbers 1 through n.  This function will need to accept an argument <code>n</code> representing the final number. Then it will need to call itself with progressively smaller values of <code>n</code> until it reaches 1. You could write the function as follows:
+
+```js
+function count(n) {
+  if (n === 1) {
+    return [1];
+  } else {
+    var numbers = count(n - 1); 
+    numbers.push(n);
+    return numbers;
+  }
+}
+```
+
+At first this is counterintuitive since the value of `n` <em>decreases</em>, but the values in the final array are <em>increasing</em>.  This happens because the push happens last, after the recursive call has returned.  At the point where `n` is pushed into the array, `count(n - 1)` has already been evaluated and returned `[1, 2, ..., n - 1]`.
+
+</section>
+
+## Instructions
+<section id='instructions'>
+We have defined a function named <code>rangeOfNumbers</code> with two parameters. The function should return an array of integers which begins with a number represented by the <code>startNum</code> parameter and ends with a number represented by the <code>endNum</code> parameter.  The starting number will always be less than or equal to the ending number.  Your function must use recursion by calling itself and not use loops of any kind.
```

Comment:
I think we need to specify what the expected result should be when both `startNum` and `endNum` are the same. It *is* evident once you calculate recursively, but may not be that clear for beginners.